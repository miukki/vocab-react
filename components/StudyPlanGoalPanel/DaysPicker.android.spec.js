/* eslint-env jest */

import 'react-native'
import React from 'react'
import DaysPicker from './DaysPicker.android'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import jsdom from 'jsdom'

jest.mock('native-base', () => {
  return { Button: 'Button' }
})

describe('description', function() {
  const dailyWordsAmounts = [5, 10, 15, 20, 25, 30, 40]

  it('render', function() {
    const tree = renderer
      .create(
        <DaysPicker
          dailyWordsAmounts={dailyWordsAmounts}
          dailyWordsAmount={20}
          changeDailyWordsAmountWithoutConfirm={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('description', function() {
    const { JSDOM } = jsdom

    const wrapper = mount(
      <DaysPicker
        dailyWordsAmounts={dailyWordsAmounts}
        dailyWordsAmount={20}
        changeDailyWordsAmountWithoutConfirm={() => {}}
      />
    )
    expect(wrapper.state().dailyWordsAmount).toBe(20)
    wrapper.setProps({ dailyWordsAmount: 40 })
    expect(wrapper.state().dailyWordsAmount).toBe(40)
  })
})
