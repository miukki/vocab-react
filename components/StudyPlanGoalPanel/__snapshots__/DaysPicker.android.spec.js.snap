// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description render 1`] = `
<View
  style={null}
>
  <RCTPicker
    items={
      Array [
        Object {
          "label": "5 words",
          "textColor": undefined,
          "value": 5,
        },
        Object {
          "label": "10 words",
          "textColor": undefined,
          "value": 10,
        },
        Object {
          "label": "15 words",
          "textColor": undefined,
          "value": 15,
        },
        Object {
          "label": "20 words",
          "textColor": undefined,
          "value": 20,
        },
        Object {
          "label": "25 words",
          "textColor": undefined,
          "value": 25,
        },
        Object {
          "label": "30 words",
          "textColor": undefined,
          "value": 30,
        },
        Object {
          "label": "40 words",
          "textColor": undefined,
          "value": 40,
        },
      ]
    }
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    selectedIndex={3}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
